v2/kafka-to-gcs/src/main/java/com/google/cloud/teleport/v2/transforms/AvroWriteTransform.java

```diff
@@ -222,17 +287,38 @@ public void setup() {
       }
     }
 
+    GenericRecord deserializeBytes(byte[] bytes, String topicName) {
+      KafkaAvroDeserializer deserializer = new KafkaAvroDeserializer(schemaRegistryClient);
+      return (GenericRecord) deserializer.deserialize(topicName, bytes);
+    }
+
+    /*
+     * This method drops the key of the KafkaRecord and deserializes the value of the KafkaRecord to
+     * GenericRecord.
+     */
     @ProcessElement
     // TODO: Add Dead letter queue when deserialization error happens.
     public void processElement(
-        @Element KafkaRecord<byte[], byte[]> kafkaRecord, OutputReceiver<GenericRecord> o) {
-      KafkaAvroDeserializer deserializer = new KafkaAvroDeserializer(schemaRegistryClient);
-      if (!useMock) {
-        o.output(
-            (GenericRecord)
-                deserializer.deserialize(kafkaRecord.getTopic(), kafkaRecord.getKV().getValue()));
-      } else {
-        o.output((GenericRecord) deserializer.deserialize(subject, kafkaRecord.getKV().getValue()));
+        @Element KafkaRecord<byte[], byte[]> kafkaRecord, MultiOutputReceiver receiver)
+        throws Exception {
+      GenericRecord genericRecord;
+      byte[] kafkaRecordValueInBytes = kafkaRecord.getKV().getValue();
+      try {
+        if (!useMock) {
+          genericRecord = deserializeBytes(kafkaRecordValueInBytes, kafkaRecord.getTopic());
+          // TODO: Remove the if condition
+          if ("SimpleMessage".equals(genericRecord.getSchema().getName())) {
```

Comment:
@Dippatel98 you can do something like this while testing this PR for kafka to BQ template